submission Homework Week3

Mehmet-Ergorgec opened the pull request:
Netlify link is created another repo. But it works.
https://nostalgic-joliot-9558ef.netlify.com/

ArcoMul commented on homework-submission/react-hyf/react-hyf-client/src/folders/week3/HomeWorkWeek3.js:
```diff
@@ -0,0 +1,57 @@
+    getUsers();
+  }, [index]);
+
+  const handleClick = e => {
```
The page is scrolling down when pressing space bar (default browser behaviour) see if you can prevent that

Mehmet-Ergorgec replied:
Yes I forgot to add e.preventDefault()

ArcoMul commented on homework-submission/react-hyf/react-hyf-client/src/folders/week3/HomeWorkWeek3.js:
```diff
@@ -0,0 +1,57 @@
+
+  const handleClick = e => {
+    if (e.keyCode === 32) {
+      setIndex(randomNum);
```
Is the value of `randomNum` really random? Often the image doesn't change when pressing space bar, can you see why?

Mehmet-Ergorgec replied:
Yes It is random. I re-checked it putting console.log(randomNum).

ArcoMul commented on homework-submission/react-hyf/react-hyf-client/src/folders/week3/HomeWorkWeek3.js:
```diff
@@ -0,0 +1,57 @@
+  const userValues = Object.values(user);
+  useEffect(() => {
+    const getUsers = async () => {
+      const res = await fetch(`https://uinames.com/api/?amount=10&&ext=photos`);
```
Would be good to show somewhere that new data is being loaded, now after pressing space it takes a bit of time and then the data suddenly changes.

Mehmet-Ergorgec replied:
It was shown only small div at the top of the window. Now it is bigger and used spinner

ArcoMul commented on homework-submission/react-hyf/react-hyf-client/src/components/week3/ProfilPicture/index.jsx:
```diff
@@ -0,0 +1,8 @@
+import './style.css';
+
+function ProfilPicture({ userValues }) {
+  return <img src={userValues[11]} alt={userValues[0]} />;
```
Don't you receive an object from the api? Why are you accessing it using an index? :)

ArcoMul commented on homework-submission/react-hyf/react-hyf-client/src/components/week3/UserInfo/index.jsx:
```diff
@@ -0,0 +1,19 @@
+      <p style={getNameStyle}>{` ${userValues[0]} ${userValues[1]}`}</p>
+      <div>
+        <p> {`👤 ${userValues[2]}`}</p>
+        <p> {`🌐 ${userValues[3]}`}</p>
```
Please show more values you get from the API as requested in the homework and shown here: https://uinames.com/

Mehmet-Ergorgec replied:
I misunderstood the homework . https://uinames.com/api/?amount=10 What I see in this link is only those information. I added picture because someone asked in Slack. So I am going to add the others.

ArcoMul commented:
Did you also update your netlify application?

Also this line, on line 11 in `homework-submission/react-hyf/react-hyf-client/src/folders/week3/HomeWorkWeek3.js`:
```js
const userValues = Object.values(user);
```
causes your data to be an array. That's why you need to access your data as an array `<p> {`👤 ${userValues[2]}`}</p>`, I don't see why it's needed to convert it to an array, please look into that :-)

ArcoMul commented:
Your other changes look good by the way! Thanks! Just this one thing left then.

Mehmet-Ergorgec commented:
Yes You are right. No need to use `Object.values()`. I changed my code and commited. Also I updated and pushed my hw4. You can see both changes from week3 and week4 hw from this link. 
https://loving-agnesi-110ced.netlify.com/

ArcoMul commented:
Super!!! 👏
